src/test/scala/ai/privado/threatEngine/go/ThreatTests.scala

```diff
@@ -0,0 +1,237 @@
+          |	ID      string `gorm:"PRIMARY KEY; UNIQUE" json:"id"`
+          |	Name    string `gorm:"type:varchar(255); NOT NULL" json:"name" validate:"required"`
+          |	Email   string `gorm:"type:varchar(255)" json:"email"`
+          |	Phone   string `gorm:"type:varchar(100); NOT NULL; UNIQUE; UNIQUE_INDEX" json:"phone" validate:"required"`
+          |	Address string `gorm:"type:text" json:"address"`
+          |	MyModel
+          |}
+          |
+          |// User Model
+          |type Student struct {
+          |	// we need to use our own custom model because the id is different of the "gorm.Model"
+          |	ID      string `gorm:"PRIMARY KEY; UNIQUE" json:"id"`
+          |	Name    string `gorm:"type:varchar(255); NOT NULL" json:"name" validate:"required"`
+          |	Email   string `gorm:"type:varchar(255)" json:"email"`
+          |	Phone   string `gorm:"type:varchar(100); NOT NULL; UNIQUE; UNIQUE_INDEX" json:"phone" validate:"required"`
+          |	Address string `gorm:"type:text" json:"address"`
+          |	MyModel
+          |}
+          |
+          |// Check if Users Model implements/match Model interface
+          |var _ Model = &User{}
+          |
+          |
+          |// FindOneByID gets a user by id
+          |func (u *User) FindOneByID(db *gorm.DB) (Model, error) {
+          |	err := db.Find(&u).Error
+          |	if err != nil {
+          |		return nil, err
+          |	}
+          |	return u, nil
+          |}
+          |
+          |
+          |// FindOneByID gets a user by id
+          |func (u *Student) FindOneByID(db *gorm.DB) (Model, error) {
+          |	err := db.Find(&u).Error
+          |	if err != nil {
+          |		return nil, err
+          |	}
+          |	return u, nil
+          |}
+          |
+          |""".stripMargin)
+
+      val result = threatEngine.processProcessingViolations(threat)
+      result should not be empty
+      result.get.policyId shouldBe "PrivadoPolicy.Storage.IsSamePIIShouldNotBePresentInMultipleTables"
+    }
+  }
+
+  "No Threat  PIIShouldNotBePresentInMultipleTables when both model have different PII" should {
+    val threat = PolicyOrThreat(
+      "PrivadoPolicy.Storage.IsSamePIIShouldNotBePresentInMultipleTables",
+      "{DataElement} was found in multiple tables",
+      "{DataElement} found in multiple tables",
+      """
+        |Avoid storing same PII in multiple tables.
+        |Reference link: https://github.com/OWASP/owasp-mstg/blob/v1.4.0/Document/0x05d-Testing-Data-Storage.md#testing-local-storage-for-sensitive-data-mstg-storage-1-and-mstg-storage-2
+        |""".stripMargin,
+      PolicyThreatType.THREAT,
+      PolicyAction.DENY,
+      DataFlow(List(), SourceFilter(Option(true), ""), List()),
+      List("**"),
+      Map[String, String](),
+      Map[String, String](),
+      "",
+      Array[String]()
+    )
+
+    "When same data-element is part of multiple table in go file" in {
```

Comment:
Is the description of this test case correct?

Reply:
We have cases such as table definition exist in Go and SQL files. This test case belong to schema definition defined in .go file.